src/core/batch/batch.ts

```diff
@@ -21,13 +24,34 @@ export type BatchFileObject = string;
 
 /**
  * createFile() takes a series of DSNP messages and returns a Batch file
- * object for publishing. This method is not yet implemented.
+ * object for publishing.
  *
- * @param events  An array of DSNPEvents to include in the batch file
- * @returns         A batch file object
+ * @param messages An array of DSNPMessage to include in the batch file
+ * @returns         A URL
  */
-export const createFile = (_events: DSNPType[]): BatchFileObject => {
-  throw NotImplementedError;
+export const createFile = async (targetPath: string, messages: DSNPMessage[]): Promise<URL | null> => {
+  if (messages.length === 0) return null;
+
+  const schema = new ParquetSchema(getSchemaFor(messages[0].dsnpType));
+
+  return putStream(targetPath, async (writeStream: ReadWriteStream) => {
+    await writeBatch(writeStream, schema, messages);
+  });
+};
+
+/**
+ * writeBatch() takes a series of DSNP messages and returns a Batch file
+ * object for publishing.
+ *
+ * @param messages An array of DSNPMessage to include in the batch file
+ * @returns        A function with
+ */
+export const writeBatch = async (writeStream: WriteStream, schema: Schema, messages: DSNPMessage[]): Promise<void> => {
```

Comment:
do we have to export this method? what would be the use case for it?

Reply:
We don't have to. It made it easy to unit test?